pkg/operator/ceph/cluster/controller.go

```diff
@@ -295,15 +384,74 @@ func (c *ClusterController) onDelete(obj interface{}) {
 	}
 }
 
-func (c *ClusterController) addFinalizer(clust *rookalpha.Cluster) error {
+func (c *ClusterController) handleDelete(cluster *cephv1alpha1.Cluster, retryInterval time.Duration) error {
+
+	operatorNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
+	retryCount := 0
+	for {
+		// TODO: filter this List operation by cluster namespace on the server side
```

Comment:
no this TODO is still pending, it is not new from this PR though.  it's from the implementation of volume clean-ups to prevent zombie RBDs from months back.